docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,453 @@
+
+* Attribute on property
+* <xref:System.Text.Json.JsonSerializerOptions.Converters> collection
+* Attribute on type
+
+The difference here is that a custom converter in the `Converters` collection overrides an attribute at the type level. The intention behind this order of precedence is to make run-time changes override design-time choices. There's no way to change the precedence.
+
+### Character escaping
+
+During serialization, `Newtonsoft.Json` is relatively permissive about letting characters through without escaping them. (That is, it doesn't replace them with `\uxxxx` where `xxxx` is the character's code point). Where it does escape them, it does so by emitting a `\` before the character (for example, `"` becomes `\"`). `System.Text.Json` escapes more characters by default to provide defense-in-depth protections against cross-site scripting (XSS) or information-disclosure attacks and does so by using the six-character sequence. `System.Text.Json` escapes all non-ASCII characters by default, so you don't need to do anything if you're using `StringEscapeHandling.EscapeNonAscii` in `Newtonsoft.Json`. `System.Text.Json` also escapes HTML-sensitive characters, by default. For information about how to override the default `System.Text.Json` behavior, see [Customize character encoding](system-text-json-how-to.md#customize-character-encoding).
+
+### Deserialization of Object properties
+
+When `Newtonsoft.Json` deserializes to `Object` properties in POCOs or in dictionaries of type `Dictionary<string, Object>`, it:
+
+* Infers the type of primitive values in the JSON payload (other than `null`) and returns the stored `string`, `long`, `double`, `boolean`, or `DateTime` as a boxed object. *Primitive values* are single JSON values such as a JSON number, string, true, false, or null.
+* Returns the stored complex values as a `JObject` or `JArray`. *Complex values* are collections of JSON key-value pairs within braces (`{}` ) or lists of values within brackets (`[]`). The properties and values within the braces or brackets can have additional properties or values.
+* Returns a `null` reference when the payload has the `null` JSON literal.
+
+`System.Text.Json` stores a boxed `JsonElement` for both primitive and complex values within the `System.Object` property or dictionary value. However, it treats `null` the same as `Newtonsoft.Json` and returns a null reference when the payload has the `null` JSON literal in it.
+
+To implement type inference for `Object` properties, create a converter like the example in [How to write custom converters](system-text-json-converters-how-to.md#deserialize-inferred-types-to-object-properties).
+
+### Maximum depth
+
+`Newtonsoft.Json` doesn't have a maximum depth limit. For `System.Text.Json` there's a default limit  of 64, and it's configurable by setting (xref:System.Text.Json.JsonSerializerOptions.MaxDepth?displayProperty=nameWithType).
+
+## Scenarios using JsonSerializer that require workarounds
```

Comment:
I don't have a sample yet, but we add a workaround for supporting reading multiple json payloads while streaming. Can you file a doc issue to track this and we can revisit it later?

Something like:
https://github.com/JamesNK/Newtonsoft.Json/issues/2125
https://github.com/dotnet/corefx/issues/39843

```C#
public class TestingClass
{
    public string MyString { get; set; }
}

var stringReader = new StringReader("{\"MyString\": \"hi\"}, {\"MyString\":\"Another payload\"}");

var jsonTextReader = new JsonTextReader(stringReader)
{
    SupportMultipleContent = true
};

var jsonSerializer = new Newtonsoft.Json.JsonSerializer();

// {"MyString":"hi"}
Console.WriteLine(JsonConvert.SerializeObject(jsonSerializer.Deserialize<TestingClass>(jsonTextReader))); 

jsonTextReader.Read();

// {"MyString":"Another payload"}
Console.WriteLine(JsonConvert.SerializeObject(jsonSerializer.Deserialize<TestingClass>(jsonTextReader))); 
```

Reply:
I already have that scenario tracked in #15805, I'll add a link there to this comment.